server/src/features/events/event.service.js

```diff
@@ -705,6 +728,54 @@ export const archiveEvent = async (eventId, user) => {
       throw new ApiError(400, "Event could not be archived due to unknown reason");
     }
   }
-
   return updatedEvent;
 };
+
+
+export const restrictAccess = async (eventId, rolesToRestrict, user) => {
+  // Authorization check
+  if (!user || user.role !== "events_office") {
+    throw new ApiError(403, "Only Events Office can restrict event access");
+  }
+
```

Comment:
```suggestion
const uniqueRoles = [...new Set(rolesToRestrict)];
```
-You don't want to validate duplicates as separate entries
-You want to validate a clean final array
-You want your “identical restrictions” comparison to work correctly
-You want to save a clean, normalized array into the database